src/main/java/org/dependencytrack/resources/v1/NotificationPublisherResource.java

```diff
@@ -78,6 +77,173 @@ public Response getAllNotificationPublishers() {
+                }
+
+                if(!jsonNotificationPublisher.getName().equals(existingPublisher.getName())) {
+                    NotificationPublisher existingNotificationPublisherWithModifiedName = qm.getNotificationPublisher(jsonNotificationPublisher.getName());
+                    if(existingNotificationPublisherWithModifiedName != null) {
+                        return Response.status(Response.Status.CONFLICT).entity("An existing publisher with the name '"+existingNotificationPublisherWithModifiedName.getName()+"' already exist").build();
+                    }
+                }
+                existingPublisher.setName(jsonNotificationPublisher.getName());
+                existingPublisher.setDescription(jsonNotificationPublisher.getDescription());
+
+                Class<?> publisherClass = Class.forName(jsonNotificationPublisher.getPublisherClass());
+
+                if (Publisher.class.isAssignableFrom(publisherClass)) {
+                    existingPublisher.setPublisherClass(jsonNotificationPublisher.getPublisherClass());
+                } else {
+                    return Response.status(Response.Status.BAD_REQUEST).entity("The class "+jsonNotificationPublisher.getPublisherClass()+" does not implement "+Publisher.class.getCanonicalName()).build();
+                }
+                existingPublisher.setTemplate(jsonNotificationPublisher.getTemplate());
+                existingPublisher.setTemplateMimeType(jsonNotificationPublisher.getTemplateMimeType());
+                existingPublisher.setDefaultPublisher(false);
+                NotificationPublisher notificationPublisherUpdated = qm.updateNotificationPublisher(existingPublisher);
+                return Response.ok(notificationPublisherUpdated).build();
+            } else {
+                return Response.status(Response.Status.NOT_FOUND).entity("The UUID of the notification publisher could not be found.").build();
+            }
+        } catch (ClassNotFoundException e) {
+            return Response.status(Response.Status.BAD_REQUEST).entity("The class "+jsonNotificationPublisher.getPublisherClass()+" cannot be found").build();
+        }
+    }
+
+    @DELETE
+    @Path("/{notificationPublisherUuid}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            value = "Deletes a notification publisher and all related notification rules",
+            code = 204
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 404, message = "The UUID of the notification publisher could not be found")
+    })
+    @PermissionRequired(Permissions.Constants.SYSTEM_CONFIGURATION)
+    public Response deleteNotificationPublisher(@ApiParam(value = "The UUID of the notification publisher to delete", required = true)
```

Comment:
This endpoint should enforce that default publishers must not be deleted. ATM this is only done by the UI not showing the *Delete* button.